spring-xd-dirt/src/main/java/org/springframework/xd/dirt/server/admin/deployment/zk/DeploymentSupervisor.java

```diff
@@ -459,6 +456,9 @@ public void takeLeadership(CuratorFramework client) throws Exception {
 				jobDeployments = instantiatePathChildrenCache(client, Paths.JOB_DEPLOYMENTS);
 				jobDeployments.start(PathChildrenCache.StartMode.BUILD_INITIAL_CACHE);
 
+				deploymentQueueForConsumer = new DeploymentQueue(client, deploymentMessageConsumer, Paths.DEPLOYMENT_QUEUE,
+						executorService);
+				deploymentQueueForConsumer.start();
```

Comment:
I am fine with moving the consumer start part before sleep. Is the reason you are suggesting this because the `PathChildrenCache` start is asynchronous and we could delay starting the consumer for some more time? If this operation were synchronous, then we don't have to worry about this I think?